main.go

```diff
@@ -44,22 +45,37 @@ func init() {
 }
 
 func main() {
+	var devLogging bool
+	var runInTestMode bool
+	var runInDemoMode bool
 	var metricsAddr string
-	var enableLeaderElection bool
-	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
-	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
-		"Enable leader election for controller manager. "+
-			"Enabling this will ensure there is only one active controller manager.")
+	var watchNamespace string
+
+	flag.BoolVar(&devLogging, "dev", false, "enable developer logging")
+	flag.StringVar(&metricsAddr, "metrics-addr", "127.0.0.1:8085",
+		"The address the metric endpoint binds to.")
+	// From CAPI point of view, BMO should be able to watch all namespaces
+	// in case of a deployment that is not multi-tenant. If the deployment
+	// is for multi-tenancy, then the BMO should watch only the provided
+	// namespace.
+	flag.StringVar(&watchNamespace, "namespace", "",
+		"Namespace that the controller watches to reconcile BMO objects.")
```

Comment:
Is it safe to add this without adding the code that we used to have to look up the current namespace if it wasn't supplied?

Reply:
The old lookup code was part of the operator-sdk library that has been dropped. AFAICT, we always set the namespace explicitly. We may want to add some logic to let us use an environment variable to do that more easily than a command line flag, for when we're running inside the cluster.

Reply:
That's fine by me as long as we comply with the CAPI requirements, that the namespace should be providable by command line, and if unset (both in command line and environment variable in that case), default to all namespaces.